### DISCUSSION TITLE
Scripting / Script Locations settings

### DISCUSSION THREAD

**Discussion by maevans12** [labs4capella/python4capella#80] [Q&A]
I've been trying to install the python4capella addon into Capella 5.1.0 and have reached a couple snags.

First, the installation instructions within the sections "Python4Capella configuration", state:
"- select the Scripting / Script Locations 
- - add /Python4Capella/sample_scripts by clicking the Add Workspace button"
I don't seem to have this setting:
![image](https://user-images.githubusercontent.com/18238582/138957929-d50a9944-98f4-446b-a8ce-73d47d5be513.png)

Am I looking in the wrong place?

This seems pretty important since if I'm going to write my own scripts, I'll need to add those projects in the same way.

Also, one of the sample scripts "export_selected_element_diagrams.py" doesn't seem to work properly (it doesn't contribute to the menu items like the comments describe) and I think not being able to add scripts may be a factor.

Can someone point me to the way to add the scripts to the script locations?

**Comment by ylussaud** [comment]
Yes you are at the right place, you should have an other menu here. You should first check that you selected every features from the update site (especially the not categorized one).
Then you can try to change the perspective on the top right corner of Capella. The menu you are looking for might be filtered in the Capella perspective. And lastly, you can activate developer capabilities, go to the preference dialog and filter on capabilities at the top left corner of the dialog. The select developer capability on the right.

Let me know if it helps.

And yes the menu will not be contributed if you don't specify where EASE need to look for scripts that can contribute menus.

**Comment by maevans12** [comment]
OK, thanks.  There is one item that didn't get installed.  It is the "EASE
UI Components".  The error during install is:

Cannot complete the install because one or more required items could not be
found.
  Software being installed: EASE UI Components (Incubation)
0.7.0.I201907161104 (org.eclipse.ease.ui.feature.feature.group
0.7.0.I201907161104)
  Missing requirement: org.eclipse.ease.ui.completions.java
0.7.0.I201906281229 requires 'osgi.bundle; org.objectweb.asm [3.3.1,8.0.0)'
but it could not be found
  Cannot satisfy dependency:
    From: EASE UI Components (Incubation) 0.7.0.I201907161104
(org.eclipse.ease.ui.feature.feature.group 0.7.0.I201907161104)
    To: org.eclipse.equinox.p2.iu; org.eclipse.ease.ui.completions.java
[0.7.0.I201906281229,0.7.0.I201906281229]

Maybe the better question is, what is the actual update site for
python4capalla?  What I did is download the .zip file from the git site,
extract that into a folder, and point Capella to that location as the
"update site".  However, it doesn't seem to be able to resolve the
dependencies.

On Wed, Oct 27, 2021 at 10:55 AM ylussaud ***@***.***> wrote:

> Yes you are at the right place, you should have an other menu here. You
> should first check that you selected every features from the update site
> (especially the not categorized one).
> Then you can try to change the perspective on the top right corner of
> Capella. The menu you are looking for might be filtered in the Capella
> perspective. And lastly, you can activate developer capabilities, go to the
> preference dialog and filter on capabilities at the top left corner of the
> dialog. The select developer capability on the right.
>
> Let me know if it helps.
>
> And yes the menu will not be contributed if you don't specify where EASE
> need to look for scripts that can contribute menus.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/labs4capella/python4capella/discussions/80#discussioncomment-1546720>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AELEY5RT6LWPTGMN2GFSCMLUJAVIVANCNFSM5GYXCROQ>
> .
>


-- 
-Matt Evans

(c) 512-517-5187

**Comment by ylussaud** [comment]
The last release can be found here:
https://github.com/labs4capella/python4capella/releases/tag/1.0.0

It looks like a dependency is missing. You can add the following Orbit update site to get this dependency:
https://download.eclipse.org/tools/orbit/downloads/drops/R20200529191137/repository

Also you will probably want to have a look at issue #77 to fix a known bug on the Capella version computation.

**Reply by maevans12** [reply]
Thanks.  

I'm not seeing that orbit has the dependency that I needed (EASE UI Components).  What I did (which seemed to work) is I went to the Eclipse Marketplace and found the Eclipse Advanced Scripting Environment (EASE) within the marketplace (https://marketplace.eclipse.org/content/eclipse-advanced-scripting-environment-ease), found the update site for it and updated EASE.  It updated components that had been installed from the python4capella update site and added other components which included the "EASE UI Components" component.  This seems to work (menus appear, I can find the "Scripting / Script Locations" tab and include the script folder like the installation said).  I also ran the export_selected_element_diagrams.py from the Export Menu and that now works well.

**Reply by maevans12** [reply]
the updated site I used for EASE is http://download.eclipse.org/ease/release/latest.

**Reply by maevans12** [reply]
It looks like I'm using EASE vs 0.8.0...

**Comment by ylussaud** [comment]
I opened issue #82 to keep track of the missing dependency problem.
